### PR TITLE
Update for new starter kits

### PR DESCRIPTION
**Pull request by jesseleite**
- [x] Add support for new starter kits.
    - [x] Polish cli messaging around starter kit license check. @jackmcdade?
- [x] Prompt to create super user at end of successful install.
- [x] Improve test coverage.
- [x] Lots of cleanup.

### CODE REVIEW

File: src/NewCommand.php [hunk 1]
```diff
@@ -28,193 +47,451 @@ protected function configure()
...
+        $details = json_decode($response->getBody(), true);
+
+        // If $details === `false`, then no product was returned and we'll consider it a free starter kit.
+        if ($details === false) {
+            return $this;
+        }
+
+        // If the returned product doesn't have a price, then we'll consider it a free starter kit.
+        if (! $details['data']['price']) {
+            return $this;
```
Comment (jasonvarga) [inline src/NewCommand.php:225]: I think this is different now. When I try to `statamic new mysite non-existent/kit` the JSON would be `{"data": false}`. I don't think `$details` is ever `false` anymore.

File: src/NewCommand.php [hunk 2]
```diff
@@ -28,193 +47,451 @@ protected function configure()
...
-            $this->createProjectCommand($repo, $directory)
-        ];
+        $this->output->write('<comment>This is a paid starter kit. If you haven\'t already, you may purchase a license at:</comment>'.PHP_EOL);
+        $this->output->write("<comment>https://statamic.com/starter-kits/{$this->starterKit}</comment>".PHP_EOL);
+
+        $helper = $this->getHelper('question');
 
-        if ($directory != '.' && $input->getOption('force')) {
+        $question = new Question('Please enter your license key: ');
+        $question->setHidden(true);
```
Comment (jasonvarga) [inline src/NewCommand.php:234]: I don't think this needs to be hidden. Kinda threw me off when I pasted in my key.

**Review by jasonvarga — CHANGES_REQUESTED** [review]

File: src/NewCommand.php (hunk 1 above)
Comment (jesseleite) [reply src/NewCommand.php]: Good catch, thanks

**Comment by jasonvarga** [comment]
Instead of a yaml file with the starter kits, can't we pull them from the marketplace?

**Comment by jackmcdade** [comment]
@jasonvarga Yes please. In fact, we should probably just point them to the Marketplace instead of trying to populate the list of them here. For now it's just a few, but in a few weeks/months there could be dozens. It'll be easier to cut it off now rather than later.

**Comment by jasonvarga** [comment]
So you want to remove this screen? Just make it a regular text input?

image

**Comment by jackmcdade** [comment]
Maybe? What options do we have that don't involve simply listing all possible kits?

**Comment by jasonvarga** [comment]
Some kind of search maybe... But that seems silly on the cli.

We could do autocompletion though.

**Comment by jackmcdade** [comment]
Maybe we do a prompt – like...

```
What do you want to do?
[0] Start with the default empty site
[1] Use a Starter Kit
```

And then if they pick `[1]`...

```
Enter the name of the Starter Kit (find them at statamic.com/stater-kits):
> autocomplete field
```

**Comment by jasonvarga** [comment]
We can do that.

**Comment by jackmcdade** [comment]
8ukb

**Comment by jesseleite** [comment]
There we go! We can do the autocomplete later though.

image

**Review by jasonvarga — APPROVED** [review]

**Review by jasonvarga — APPROVED** [review]

**Pull request merged**